src/vm/customattribute.cpp

```diff
@@ -165,11 +165,11 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         ULONG length = pCaVal->arr.length;
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
-        if (length == (ULONG)-1)
-            return gc;
-        
-        gc.array = (CaValueArrayREF)AllocateValueSzArray(MscorlibBinder::GetClass(CLASS__CUSTOM_ATTRIBUTE_ENCODED_ARGUMENT), length);
-        CustomAttributeValue* pValues = gc.array->GetDirectPointerToNonObjectElements();
+        if (length != (ULONG)-1)
+        {
+            gc.array = (CaValueArrayREF)AllocateValueSzArray(MscorlibBinder::GetClass(CLASS__CUSTOM_ATTRIBUTE_ENCODED_ARGUMENT), length);
+            CustomAttributeValue* pValues = gc.array->GetDirectPointerToNonObjectElements();
+        }
 
         for (COUNT_T i = 0; i < length; i ++)
```

Comment:
This loop shouldn't be executed if the array is null (yes, it conveniently early-out's anyway since the "length" is -1 but this is fragile and unintuitive.) If you follow my suggestion of retaining the early-out structure, then this issue will be moot.